lang_python/parsing/AST_python.ml

```diff
@@ -319,6 +319,12 @@ and param_pattern =
 *)
 and type_ = expr
 
+(* This is purely for LSIF type parsing, which is needed since Python types
```

Comment:
pfff is a repository that is independent of semgrep. It's used in many of my other projects.
Reading this comment would not help the reader understand what this type is for.
You have to put yourself in the shoe of someone else, or even you in 6 months reading this comment.

The reader has no idea what LSIF is, and why we need that.
You can be quick like "This is used in semgrep-proprietary, a variant of semgrep, where
we read type information about Python code in LSIF files (see lsif.org or whatever)"